R/environment.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python
+import json
+import locale
+import os
+import re
+import subprocess
+import sys
+
+
+version_re = re.compile(r'\d+\.\d+(\.\d+)?')
+exec_dir = os.path.dirname(sys.executable)
+
+class EnvironmentException(Exception):
+    pass
+
+
+def detect_environment(dirname):
+    """Determine the python dependencies in the environment.
+
+    If requirements.txt exists in the notebook directory,
+    its contents will be used. Otherwise, the results
+    of `pip freeze` will be used.
+
+    Returns a dictionary containing the package spec filename
+    and contents if successful, or a dictionary containing 'error'
+    on failure.
+    """
+    result = (output_file(dirname, 'requirements.txt', 'pip') or
+              pip_freeze(dirname))
+
+    if result is not None:
+        result['python'] = get_python_version()
+        result['pip'] = get_version('pip')
```

Comment:
Could you do:

```
import pip
pip.__version__
```

Reply:
As of pip 10.0, the pip module can no longer be imported (pip issue 5081) but we could probably run `python -m pip --version`.